Homework/Astro.cs

```diff
@@ -0,0 +1,14 @@
+﻿namespace Homework
+{
+    internal class Astro
+    {
+        internal string sunrise {  get; set; }
+        internal string sunset { get; set; }
+        internal string moonrise { get; set; }
+        internal string moonset { get; set; }
+        internal string moon_phase { get; set; }
```

Comment:
Tady pridam informaci k nazyvani promennych DTO objektu urcenych k (de)serializaci. Idealne by jsme chteli dodrzovat naming konvenci, co ma c# (takze tato property by se nejmenovala `moon_phase`, ale `MoonPhase`). 

Chapu, ze ti to bez tohohle pojmenovani nebude spravne deserializovat. Jde o to, ze v jsonu, ktery to apicko vraci jsou nektere property objektu nazvany trochu jinak, nez je konvence v nazyvani properties v c#, pak se pri deserializaci nedokaze urcit, ke ktere property se ma hodnota priradit pokud to nematchuje 1:1

Protoze vetsinou to nematchuje 1:1, tak se pomoci atributu doplnuje informace o tom, jak se to ma (at uz pro deserializaci, nebo serializaci) jmenovat.

Takto by to vypadalo napsane spravne, aby deserializer vedel, jak to spravne namapovat:
```suggestion
        [JsonPropertyName("moon_phase")]
        public string MoonPhase { get; set; }
```

Reply:
A vsimni si take, ze jsem ti zmenil v navrhu pristupovy modifikator z `internal` na `public`. To je ten duvod, proc ti to nefunguje.

Deserializer se snazi najit v objektu public property a ty pomoci setteru naplnit. Tim, ze ta property neni public, tak on ji ignoruje.

Konceptualne to dava smysl, protoze tim DTO objektem popisujes nejaky contract/smlouvu, kterou opisujes od externi sluzby. Ta je verejna, neni to zadne tajemstvi a tak tedy ani neni nutno jim omezovat v tvem projektu pristup kdyz je naplnujes z externich dat.

Nasel jsem nejakou zadost na spravce knihovny, aby podporu pro deserializaci do ne-public promennych pridali, ale primo autori se vyjadrili, ze to delat neplanuji 😄 
https://github.com/dotnet/runtime/issues/31511

Reply:
A tady nejaky clanek k tem atributum, pomoci kterych reknes (de)serializeru, jak namapovat:
en: https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/customize-properties
cz: https://learn.microsoft.com/cs-cz/dotnet/standard/serialization/system-text-json/customize-properties